drivers/amazonec2/amazonec2.go

```diff
@@ -1546,6 +1549,25 @@ func (d *Driver) getRegionZone() string {
 	return d.Zone
 }
 
+// buildEC2Tags accepts a string of tagGroups (in the format of 'key1,value1,key2,value2')
+// and returns a slice of ec2.Tag's which can be applied to various ec2 resources.
+func buildEC2Tags(tagGroups string) []*ec2.Tag {
+	var tags []*ec2.Tag
+	if tagGroups != "" {
+		t := strings.Split(tagGroups, ",")
+		if len(t) > 0 && len(t)%2 != 0 {
+			log.Warnf("Tags are not key value in pairs. %d elements found", len(t))
+		}
```

Comment:
Same comment as above: if you know what the length of a slice will be, then set the capacity to avoid extra allocations.
```suggestion
		}
		tags = make([]*ec2.Tag, 0, len(t)/2)
```